Yubico.Core/src/Yubico/PlatformInterop/NativeMethods.cs

```diff
@@ -24,6 +24,11 @@ internal static partial class NativeMethods
         private const string MacDlLib = "libdl.dylib";
         private const string LinuxDlLib = "libdl.so";
 
+        static NativeMethods()
+        {
+            Libraries.EnsureInitialized();
+        }
+
```

Comment:
Why not embed this in the `Libraries.Net47.cs` file? Do we need the level of indirection here? Couldn't we just make loading NativeShims part of the static constructor of the Libraries class?

```csharp
internal static Libraries()
{
   // Wrap in the try/catch like you already do. Omitted for brevity.
   if (_nativeShims != null)
   {
       return;
   }

   _nativeShims = UnmanagedDynamicLibrary.Open(NativeShimsPath);
}

Reply:
I tried that first, but it wouldn't work. I did some research and it could be that the DllImport is executed at an earlier stage that static constructors. Then you might ask, well then, why does it work when you put the EnsureInitialized in the NativeMethods constructor? 
Answer is, I don't know. Could it be because of higher importance since the DllImport and the constructors are in the same class? I even tried throwing an exception in the Libraries constructor just to see if that would get thrown before the DllImport fails, which it didnt. So.. yeah. Weird one.

Reply:
😁 Yeah, I figured it may have been something like that. A mystery for another time perhaps. OK to resolve thread.